### ISSUE TITLE
Enum definition based on condition

### ISSUE THREAD

**Issue by haarisr** [microsoft/pyright#9719]
**Describe the bug**
Running the below code works in python but pyright throws an error.  Pyright does not throw an error if `x=False`

**Code or Screenshots**
```python
from enum import IntEnum

x: bool = True
if x:

    class Foo(IntEnum):
        A = 1
        B = 2
        C = 3
else:

    class Foo(IntEnum):
        A = 1
        B = 2
        C = 3
        D = 4
        E = 5


res = Foo.A

```

```bash
pyright foo.py 
/home/haarisr/work/kato/python/foo.py
  /home/haarisr/work/kato/python/foo.py:6:11 - error: Class declaration "Foo" is obscured by a declaration of the same name (reportRedeclaration)
1 error, 0 warnings, 0 informations
```

**VS Code extension or command-line**
Latest pyright on CLI

**Comment by erictraut** [comment]
Pyright is working as intended here, so this isn't a bug.

You are defining the same symbol twice in an incompatible manner.

A `class` statement is considered a type declaration. For more details about type declarations, refer to [this documentation](https://microsoft.github.io/pyright/#/type-concepts).

In contrast to the runtime, which executes only one branch of an `if` statement, a static analyzer must consider all possible code paths.

**Comment by haarisr** [comment]
```python3
from enum import IntEnum

x: bool = False
if x:

    class Foo(IntEnum):
        A = 1
        B = 2
        C = 3
else:

    class Foo(IntEnum):
        A = 1
        B = 2
        C = 3
        D = 4
        E = 5


res = Foo.A
```

```bash
pyright foo.py
0 errors, 0 warnings, 0 informations
```


Then this should also be an error?
